k8s/jiva/snapshot-cleanup.sh

```diff
@@ -0,0 +1,138 @@
+    echo "Do you want to continue (Y/N)"
+    read access
+    if [ $access == "N" ]; then
+    exit 1;
+    fi
+}
+
+# spin prints ('|','/','-','\','|') in cyclic order while snapshot deletion is in progres.
+spin()
+{
+    while true
+    do
+      stat /proc/$pid > /dev/null
+      if [ $? -ne 0 ]
+      then
+          break
+      fi
+      printf "\b${sp:i++%${#sp}:1}"
+      sleep 1
+    done
+}
+
+# delete_jiva_snapshot deletes jiva-snapshots(based on the arguments passed for number of snapshots to be deleted) using jivactl 
+delete_jiva_snapshot()
+{   
+    warning
+    min_required_snapshot=4
+    count=0
+    
+    snapshot_list_cmd=$(kubectl exec -it $ctrl_pod_name -n $pvc_namespace -- jivactl snapshot ls)
+    if [ "$?" != "0" ]; then
+       exit 1;
+    fi
+
+    snapshot_number_cmd='echo $snapshot_list_cmd | tr " " "\n" | grep -v ID | wc -l'
+    snapshot_name_cmd='echo $snapshot_list_cmd | tr " " "\n" | grep -v ID | tail -1'
+
+    if [ $(eval $snapshot_number_cmd) -le $((number_of_snapshot-1)) ]; then
+        echo "Error: You have requested to delete. $number_of_snapshot. There are only $(($(eval $snapshot_number_cmd) - $min_required_snapshot)). snapshot that can be deleted snapshots available on this volume. Please re-run this command with by specifying the number of snapshots to be deleted as $(($(eval $snapshot_number_cmd) - $min_required_snapshot)) or less."
+        exit 1 ;
+    else
+            if [ $(eval $snapshot_number_cmd) -gt $(($min_required_snapshot + $number_of_snapshot)) ]; then
+                block_service
+                echo "Deleting snapshots"
+                delete_snap > /dev/null 2&>1 &
+                pid=$!
+                spin
+                cat log.txt
+                unblock_service
+                restart_ctrl
+                rm -rf log.txt
+            else
+                echo "Error: You have requested to delete. $number_of_snapshot. There are only $(($(eval $snapshot_number_cmd) - $min_required_snapshot)). snapshot that can be deleted snapshots available on this volume. Please re-run this command with by specifying the number of snapshots to be deleted as $(($(eval $snapshot_number_cmd) - $min_required_snapshot)) or less." ;
+            fi
+        fi
+}
+
+# block_service changes target port value with different value
+# To restrict io's to happen
+block_service()
+{
+    kubectl patch svc $ctrl_service_name -n $pvc_namespace --type merge -p "$(cat tmp.json)"  
+}
+
+
+# unblock_service restores the actual target port value to allow io's
+unblock_service()
+{
+    sed -i 's/50000/3260/g' tmp.json
+    kubectl patch svc $ctrl_service_name -n $pvc_namespace --type merge -p "$(cat tmp.json)"
```

Comment:
do a restart controller before unblocking

Reply:
@vishnuitta I made this change.